src/analysis/seminorm.lean

```diff
@@ -275,6 +275,26 @@ variables (p : seminorm 𝕜 E) (c : 𝕜) (x y : E) (r : ℝ)
 protected lemma smul : p (c • x) = ∥c∥ * p x := p.smul' _ _
 protected lemma triangle : p (x + y) ≤ p x + p y := p.triangle' _ _
 
+instance : has_add (seminorm 𝕜 E) :=
+  { add := λ p q, { to_fun := λ x, p x + q x,
+    smul' := λ a x, by rw [p.smul, q.smul, mul_add],
+    triangle' := λ _ _, has_le.le.trans_eq (add_le_add (p.triangle _ _) (q.triangle _ _))
+      (add_add_add_comm _ _ _ _) } }
+
+lemma coe_add (p q : seminorm 𝕜 E) : ⇑(p + q) = p + q := rfl
+
+instance : has_scalar nnreal (seminorm 𝕜 E) :=
```

Comment:
I think this is true more generally for
```suggestion
instance [algebra R nnreal] [has_scalar R ℝ] [is_scalar_tower R nnreal ℝ] : has_scalar R (seminorm 𝕜 E) :=
```
Which is nice because it gives you the nat action

Reply:
Do we actually need the nat action? for getting `[add_monoid]` we could also use `function.injective.add_comm_monoid`.

Reply:
Here's the version I'm thinking of:
```
/-- Any action on `ℝ` which factors through `ℝ≥0` applies to a seminorm. -/
instance {R} [has_scalar R ℝ] [has_scalar R ℝ≥0] [is_scalar_tower R ℝ≥0 ℝ] :
  has_scalar R (seminorm 𝕜 E) :=
{ smul := λ r p,
  { to_fun := λ x, r • p x,
    smul' := λ _ _, begin
      simp only [←smul_one_smul ℝ≥0 r (_ : ℝ), nnreal.smul_def, smul_eq_mul],
      rw [p.smul, mul_left_comm],
    end,
    triangle' := λ _ _, begin
      simp only [←smul_one_smul ℝ≥0 r (_ : ℝ), nnreal.smul_def, smul_eq_mul],
      exact (mul_le_mul_of_nonneg_left (p.triangle _ _) (nnreal.coe_nonneg _)).trans_eq (mul_add _ _ _),
    end } }

lemma coe_smul {R} [has_scalar R ℝ] [has_scalar R ℝ≥0] [is_scalar_tower R ℝ≥0 ℝ]
  (r : R) (p : seminorm 𝕜 E) : ⇑(r • p) = r • p := rfl
```

Reply:
> Do we actually need the nat action? for getting `[add_monoid]` we could also use `function.injective.add_comm_monoid`.

If you do that, then the `coe_smul` lemma won't work for `nsmul`. If you do it the way I suggest, it does